lib/Chat/ReactionManager.php

```diff
@@ -0,0 +1,172 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Chat;
+
+use OCA\Talk\Exceptions\ReactionAlreadyExistsException;
+use OCA\Talk\Model\Attendee;
+use OCA\Talk\Participant;
+use OCA\Talk\Room;
+use OCA\Talk\Service\ParticipantService;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\Comments\IComment;
+use OCP\Comments\ICommentsManager;
+use OCP\Comments\NotFoundException;
+use OCP\IL10N;
+use OCP\Notification\IManager as INotificationManager;
+
+class ReactionManager {
+	/** @var ICommentsManager|CommentsManager */
+	private $commentsManager;
+	/** @var IL10N */
+	private $l;
+	/** @var MessageParser */
+	private $messageParser;
+	/** @var INotificationManager */
+	private $notificationManager;
+	/** @var Notifier */
+	private $notifier;
+	/** @var ParticipantService */
+	private $participantService;
+	/** @var ITimeFactory */
+	protected $timeFactory;
+
+	public function __construct(CommentsManager $commentsManager,
+								IL10N $l,
+								MessageParser $messageParser,
+								INotificationManager $notificationManager,
+								Notifier $notifier,
+								ParticipantService $participantService,
+								ITimeFactory $timeFactory) {
+		$this->commentsManager = $commentsManager;
+		$this->l = $l;
+		$this->messageParser = $messageParser;
+		$this->notificationManager = $notificationManager;
+		$this->notifier = $notifier;
+		$this->participantService = $participantService;
+		$this->timeFactory = $timeFactory;
+	}
+
+	public function addReactionMessage(Room $chat, Participant $participant, IComment $parentMessage, string $reaction): IComment {
+		try {
+			// Check if the user already reacted with the same reaction
+			$comment = $this->commentsManager->getReactionComment(
+				(int) $parentMessage->getId(),
+				$participant->getAttendee()->getActorType(),
+				$participant->getAttendee()->getActorId(),
+				$reaction
+			);
+			throw new ReactionAlreadyExistsException();
+		} catch (NotFoundException $e) {
+		}
+
+		$comment = $this->commentsManager->create(
+			$participant->getAttendee()->getActorType(),
+			$participant->getAttendee()->getActorId(),
+			'chat',
+			(string) $chat->getId()
+		);
+		$comment->setParentId((string) $parentMessage->getId());
+		$comment->setMessage($reaction);
+		$comment->setVerb('reaction');
+		$this->commentsManager->save($comment);
+
+		$this->notifyIfNecessary($chat, $comment, $parentMessage);
+		return $comment;
+	}
+
+	private function notifyIfNecessary(Room $chat, IComment $reactionInstance, IComment $parentMessage): void {
+		if ($chat->getType() !== Room::TYPE_ONE_TO_ONE) {
+			return;
+		}
```

Comment:
this was a misunderstanding.

We also want notifications, if users are:
* on `NOTIFY_ALWAYS` or
* on `NOTIFY_DEFAULT` and
    * room is `TYPE_ONE_TO_ONE` or
    * getDefaultGroupNotification is `NOTIFY_ALWAYS`

Basically similar to `OCA\Talk\Chat\Notifier::notifyOtherParticipant`